atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -650,13 +674,167 @@ def set_build_image(self):
             raise RuntimeError("Build object is malformed, failed to fetch buildroot image")
 
         if build_kind == 'DockerImage':
-            self.config_kwargs['build_image'] = build_name
+            return build_name
         else:
             raise RuntimeError("Build kind isn't 'DockerImage' but %s" % build_kind)
 
+    def process_image_from(self, image_from):
+        build_image = None
+        imagestream = None
+
+        if image_from['kind'] == 'DockerImage':
+            build_image = image_from['name']
+        elif image_from['kind'] == 'ImageStreamTag':
+            imagestream = image_from['name']
+        else:
+            raise UnknownKindException
+
+        return build_image, imagestream
+
+    def set_build_image(self, platforms):
+        """
+        Overrides build_image for worker, to be same as in orchestrator build
+        """
+        current_platform = platform.processor()
+        orchestrator_platform = current_platform if current_platform else 'x86_64'
+        status = get_build_json().get("status")
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        need_manifest_list = not (orchestrator_platform in platforms and len(platforms) == 1)
+        use_same_image = False
+        current_buildimage = self.get_current_buildimage()
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
+
+        # get image build from build metadata, which is set for direct builds
+        else:
+            metadata = get_build_json().get("metadata")
+
+            if 'from' in metadata['annotations']:
+                scratch_from = json.loads(metadata['annotations']['from'])
+
+                try:
+                    build_image, imagestream = self.process_image_from(scratch_from)
+                except UnknownKindException:
+                    raise RuntimeError("Build annotation has unknown 'kind' %s" %
+                                       scratch_from['kind'])
+            elif not need_manifest_list:
```

Comment:
Should this be the first check this function does? If it doesn't need manifest list, just use the same image and bail out?

This will also ensure that this plugin does not always require an OpenShift session, which is really important for backwards compatibility.

Reply:
well not really, IF there is manifest list used, I'd like to actually verify that we are using correct digest from it

Reply:
@rcerven pointed out that the story for this (OSBS-5221) actually calls out for always making this check. I think this was an oversight. @twaugh, think we should only bother with this new behavior if manifest list is actually required (at least one worker build uses different arch than orchestrator build) ?

I'm concerned in making the new plugin parameters for building openshift session required for all cases.

Reply:
@lcarva yes, I agree. If we do not need the manifest list (because there is a single worker, for the same architecture as the orchestrator), let's skip the additional processing.

Reply:
ack